letsencrypt-auto-source/tests/centos6_tests.sh

```diff
@@ -4,75 +4,90 @@ yum update -y > /dev/null
 yum install -y centos-release-scl > /dev/null
 yum install -y python27 > /dev/null 2> /dev/null
 
+LE_AUTO_PRE_0_37="certbot/letsencrypt-auto-source/letsencrypt-auto_pre_0_37"
 LE_AUTO="certbot/letsencrypt-auto-source/letsencrypt-auto"
 
 # we're going to modify env variables, so do this in a subshell
 (
 source /opt/rh/python27/enable
 
 # ensure python 3 isn't installed
-python3 --version 2> /dev/null
+python3 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "Python3 is already installed."
+  echo "ERROR: Python3 is already installed."
   exit 1
 fi
 
 # ensure python2.7 is available
-python2.7 --version 2> /dev/null
+python2.7 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -ne 0 ]; then
-  error "Python3 is not available."
+  echo "ERROR: Python2.7 is not available."
   exit 1
 fi
 
 # bootstrap, but don't install python 3.
 "$LE_AUTO" --no-self-upgrade -n > /dev/null 2> /dev/null
 
 # ensure python 3 isn't installed
-python3 --version 2> /dev/null
+python3 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "letsencrypt-auto installed Python3 even though Python2.7 is present."
+  echo "ERROR: letsencrypt-auto installed Python3 even though Python2.7 is present."
   exit 1
 fi
 
-echo ""
 echo "PASSED: Did not upgrade to Python3 when Python2.7 is present."
 )
 
 # ensure python2.7 isn't available
-python2.7 --version 2> /dev/null
+python2.7 --version > /dev/null 2> /dev/null
 RESULT=$?
 if [ $RESULT -eq 0 ]; then
-  error "Python2.7 is still available."
+  error "ERROR: Python2.7 is still available."
   exit 1
 fi
 
 # Skip self upgrade due to Python 3 not being available.
 if ! "$LE_AUTO" 2>&1 | grep -q "WARNING: couldn't find Python"; then
-  echo "Python upgrade failure warning not printed!"
+  echo "ERROR: Python upgrade failure warning not printed!"
   exit 1
 fi
 
-# bootstrap, this time installing python3
+# add SCL rh-python36 to path for proper python 3.x resolution
+PATH="/opt/rh/rh-python36/root/usr/bin:$PATH"
```

Comment:
Yes, it is to correctly resolve python interpreter in the next call. I was initially concerned by the recommended way of calling `source scl_source enable rh-python36` because this could leak environment variable alteration to the caller shell of `certbot-auto`. But in fact it is not the case.

So I agree it is better to use `source scl_source enable rh-python36`.